frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/homepage/MyTaskAssignmentsPortlet.java

```diff
@@ -52,8 +76,36 @@ public void doView(RenderRequest request, RenderResponse response) throws IOExce
             log.error("Could not fetch your moderations from backend", e);
         }
 
-        request.setAttribute(PortalConstants.MODERATION_REQUESTS,  CommonUtils.nullToEmptyList(openModerations));
+        JSONArray jsonModerations = getModerationData(openModerations);
```

Comment:
from line 79 - 109 are duplicated in the file `MyTaskSubmissionsPortlet.java` as well. Is it possible to move it to a common place.